lib/algoliasearch-rails.rb

```diff
@@ -530,28 +475,30 @@ def algolia_reindex(batch_size = AlgoliaSearch::IndexSettings::DEFAULT_BATCH_SIZ
         next if algolia_indexing_disabled?(options)
         next if options[:replica]
 
+        algolia_ensure_init(options, settings)
+        index_name = algolia_index_name(options)
+
         # fetch the master settings
-        master_index = algolia_ensure_init(options, settings)
-        master_settings = master_index.get_settings rescue {} # if master doesn't exist yet
+        master_settings = AlgoliaSearch.client.get_settings(index_name).to_hash rescue {} # if master doesn't exist yet
         master_exists = master_settings != {}
-        master_settings.merge!(JSON.parse(settings.to_settings.to_json)) # convert symbols to strings
+        master_settings.merge!(settings.to_hash)
 
         # remove the replicas of the temporary index
         master_settings.delete :replicas
         master_settings.delete 'replicas'
 
         # init temporary index
-        src_index_name = algolia_index_name(options)
-        tmp_index_name = "#{src_index_name}.tmp"
+        tmp_index_name = "#{index_name}.tmp"
         tmp_options = options.merge({ :index_name => tmp_index_name })
         tmp_options.delete(:per_environment) # already included in the temporary index_name
         tmp_settings = settings.dup
 
         if options[:check_settings] == false && master_exists
-          AlgoliaSearch.client.copy_index!(src_index_name, tmp_index_name, { scope: %w[settings synonyms rules] })
-          tmp_index = SafeIndex.new(tmp_index_name, !!options[:raise_on_failure])
-        else
-          tmp_index = algolia_ensure_init(tmp_options, tmp_settings, master_settings)
+          task_id = AlgoliaSearch.client.operation_index(
+            index_name,
+            Algolia::Search::OperationIndexParams.new(operation: "copy", destination: tmp_index_name, scope: %w[settings synonyms rules])
```

Comment:
you could use
```suggestion
            Algolia::Search::OperationIndexParams.new(operation: Algolia::Search::OperationType::COPY, destination: tmp_index_name, scope: %w[settings synonyms rules])
```
but it's not much more pretty